scripts/main.py

```diff
@@ -0,0 +1,232 @@
+import os
+import time 
+import pickle
+import random
+import numpy as np
+import matplotlib.pyplot as plt
+import torch
+from multiprocessing import Pool
+import vboc.model as models
+from vboc.parser import Parameters, parse_args
+from vboc.abstract import SimDynamics
+from vboc.controller import ViabilityController
+from vboc.learning import NeuralNetwork, RegressionNN, plot_viability_kernel
+
+
+def computeDataOnBorder(N_guess, test_flag=0):
+    valid_data = np.empty((0, model.nx))
+    controller.resetHorizon(N_guess)
```

Comment:
I don't know if it is safe to share the same `controller` object amoung the different processes. In the past I've had issues with this.

Reply:
`Pool` is safe, since the changes (of a single process) on shared objects are not seen by other processes.
I attach a test with a fixed seed as proof. 
[main_seed.zip](https://github.com/idra-lab/VBOC/files/15008627/main_seed.zip)